### ISSUE TITLE
Sort methods alphabetically in solgen

### ISSUE THREAD

**Issue by therealbytes** [concrete-eth/concrete-geth#68]
# Rationale

Right now, in `concrete/codegen/solgen/solgen.go:generateSolidityLibrary`, ABI methods are read directly from a `map[string]abi.Method` mapping and, therefore, the ordering is randomized. This is undesirable as it makes it so that running solgen multiple times results outputs with methods in different orders.

# Implementation

Instead or reading methods from the mapping directly, initialize an `OrderedMap` with `ABI.Methods`, sort the map alphabetically by method name, then read from the sorted map.

**Comment by EmmanuelAR** [comment]
Hey @therealbytes can I work on this ? I will follow the instructions, also I will let u know the questions, do u have a telegram chat ?

I will...

1. **Extract Method Names**: 
   - Create a list of all method names from the `map[string]abi.Method`.
   - This step involves iterating over the keys of the map and storing them in a list.

2. **Sort Method Names**:
   - Sort the list of method names alphabetically.
   - Use an appropriate sorting algorithm or utility that sorts strings in alphabetical order.

3. **Access Methods in Order**:
   - Iterate over the sorted list of method names.
   - For each method name in the sorted list, access the corresponding method from the original map.

**Comment by Shoetan** [comment]
@therealbytes  I would love to take on this issue and I believe the way to go about it is 

1. Collect the method name from Abi.methods.
2.  sort the method name alphabetically.
3. Process method in sorted order

Below is how I intend to collect the method name and sort them before proceeding to process method in sorted order.
```
        methodNames := make([]string, 0, len(ABI.Methods))
	for name := range ABI.Methods {
		methodNames = append(methodNames, name)
	}
	sort.Strings(methodNames)
```

**Comment by Jemiiah** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/concrete).

### My background and how it can be leveraged
i would love to work on this issue ser if i may be given the chance

**Comment by fabrobles92** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/concrete).

### My background and how it can be leveraged
@therealbytes If not assigned I'd love to work on this issue :D

**Comment by therealbytes** [comment]
Closed by #74
